From 15b3c591abd3b0e4334f9fba19a9b75330b33c26 Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@ubuntu.com>
Date: Sat, 15 Feb 2014 22:47:02 +0000
Subject: Fix infinite loop parsing vim commands

This happens if a non-alphanumeric character other than whitespace or '-' is
found before the first alphanumeric character after 'command'.

Bug: http://sourceforge.net/tracker/index.php?func=detail&aid=3214129&group_id=6556&atid=106556
Bug-Ubuntu: https://bugs.launchpad.net/bugs/736367
Forwarded: no
Last-Update: 2011-03-17

Patch-Name: vim-command-loop.patch
---
 vim.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/vim.c b/vim.c
index 4e6fba8..d17a1ba 100644
--- a/vim.c
+++ b/vim.c
@@ -405,7 +405,9 @@ static boolean parseCommand (const unsigned char *line)
 			while (*cp && !isspace ((int) *cp))
 				++cp; 
 		}
-	} while ( *cp &&  !isalnum ((int) *cp) );
+		else
+			break;
+	} while ( *cp );
 
 	if ( ! *cp )
 	{
